python: do not double free seqnum and template_options

alltilla opened the pull request:
When these two objects are passed to `PyObject_SetAttrString()`, their reference counter is incremented. In the next line, we drop our own reference to them with `Py_DECREF()`, but still keep track of them to free it later in `_py_free_bindings()`.

It is not needed, the python lib will take care of the reference, that was added in `PyObject_SetAttrString().

Because we do not need these objects in our code, it is the easiest, to not store them in our struct.

Signed-off-by: Attila Szakacs <attila.szakacs@oneidentity.com>

kira-syslogng commented:
Build SUCCESS

MrAnno commented on modules/python/python-dest.c:
```diff
@@ -405,8 +405,6 @@ _py_init_bindings(PythonDestDriver *self)
   g_ptr_array_add(self->py._refs_to_clean, self->py.flush);
   g_ptr_array_add(self->py._refs_to_clean, self->py.send);
-  g_ptr_array_add(self->py._refs_to_clean, self->py.log_template_options);
-  g_ptr_array_add(self->py._refs_to_clean, self->py.seqnum);
```
In case we store these object in `PythonDestDriver`, we should not decrement our reference counter after setatttr, because the reference added by `PyObject_SetAttrString()` can easily be decremented on the Python side by assigning a new value to that attribute. `pyseqnum` and `pylog_template_options` would point to invalid memory in that case.

I checked and we don't really use `self->py.log_template_options` and `self->py.seqnum` after the setattr call, so I think we could just remove these refs from the C struct.

alltilla replied:
Yes-yes, good point!

Let's just make them local variables then!

alltilla replied:
Fixed!

MrAnno reviewed (commented):
Nice catch!

kira-syslogng commented:
Build SUCCESS

alltilla commented:
Updated: added `-DCMAKE_BUILD_TYPE=Debug` to `CMAKE_FLAGS`, to have debug symbols.

Test run with a manually generated corefile: https://github.com/alltilla/syslog-ng/actions/runs/554768804

kira-syslogng commented:
Build SUCCESS
